### PR TITLE
PORT: use SWI-cpp2.h

### PR DESCRIPTION
**Pull request by kamahen**
The first commit has minimal changes.
If I have time, I'll modify the code to use SWI-cpp2.h the way it was intended to be used (e.g., use PlCheckFail() or PlEx()).

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -4,29 +4,28 @@
...
+SWIPL_CFLAGS+=-I. -std=c++17 -O2 -g  # -gdwarf-2 -g3
 
 all:	$(SOBJ)
 
-
 $(SOBJ): $(OBJ)
-	mkdir -p $(PACKSODIR)
-	$(LD) $(ARCH) $(LDSOFLAGS) -o $@ $(OBJ) $(LIBS) $(SWISOLIB)
+	mkdir -p $(SWIPL_MODULE_DIR)
+	$(SWIPL_LD) $(SWIPL_MODULE_LDFLAGS) -o $@ $(OBJ) $(LIBS) $(SWIPL_MODULE_LIB)
```
Comment (kamahen) [inline Makefile.in:19]: I removed $(ARCH) because it doesn't seem to be a pre-defined variable.

**Comment by JanWielemaker** [comment]
Can't verify this right now.    Unfortunately it appears geos 3.12.2 (Fedora 40) dropped geos/geom/CoordinateArraySequence.h called from space.cc:37.    This applies to the version both before and this PR :cry: 

As a hole, this package keeps compatibility issues.   I started a new package that merely captures libspatialindex through the C interface which appears much more stable.    Didn't finish this yet and I don't know whether I'll ever finish it, given that the project for which this was meant has been abandoned.   The space package does have users and has been "fixed" several times by the SWI-Prolog Discourse user @swi and some others.

Would be nice if this could be resolved ...

**Comment by kamahen** [comment]
What do you suggest I do?

The only reason I worked on this is that you had mentioned (2023-04-20, private email) that:

> ... the Docker image currently uses 4 C++ extensions that do not work on 32-bit ARM because the old C++ interface doesn't work there: space, rocksdb, hdt and rserve_client.

and later:
> I'm unsure about the value of space.   I'm creating a new wrapper for libspatialindex.   The others should in due time be updated.  Notably rserve_client is useful (connects R to SWISH).

So, for making the docker image, I suspect that there's not much harm in merging my PR, assuming that it fixes the 32-bit ARM problem(s).

**Comment by JanWielemaker** [comment]
For now I simply merge this.   Will be verified when I try to build the docker images.   We'll fix it from there if necessary.   In due time it seems this code needs another update to deal with the libgeos changes :cry:

**Pull request merged**